src/main/scala/mesosphere/marathon/core/scheduler/SchedulerPluginRunner.scala

```diff
@@ -0,0 +1,24 @@
+package mesosphere.marathon
+package core.scheduler
+
+import mesosphere.marathon.plugin.RunSpec
+import mesosphere.marathon.plugin.scheduler.SchedulerPlugin
+import org.apache.mesos.Protos.Offer
+
+/**
+  * Runs external scheduler plugins
+  *
+  * @author Dmitry Tsydzik.
+  * @since 6/15/17.
+  */
+object SchedulerPluginRunner {
+  /**
+    * Is populated by [[mesosphere.marathon.core.scheduler.SchedulerPluginConfigurer]]
+    */
+  var plugins: List[SchedulerPlugin] = List.empty
```

Comment:
`ResourceMatcher.matchResources(..)` is called by `InstanceOpFactoryImpl` where we do have the plugin manager. At this point we could either pass a list of `SchedulerPlugin` or you could hide this list behind trait abstracting this list. Would this be an option?